crates/codegen/syntax_templates/src/shared/cst.rs

```diff
@@ -1,167 +1,62 @@
-use std::cmp::{max, min};
-use std::ops::Range;
 use std::rc::Rc;
 
 use serde::Serialize;
 
-use super::kinds::*;
-use super::language::{TextPosition, TextRange};
+use super::{
+    kinds::{RuleKind, TokenKind},
+    text_index::TextIndex,
+};
 
 #[derive(Clone, Debug, PartialEq, Eq, Serialize)]
-pub enum Node {
-    Rule {
-        kind: RuleKind,
-        range: TextRange,
-        children: Vec<Rc<Node>>,
-    },
-    Token {
-        kind: TokenKind,
-        range: TextRange,
-        #[serde(skip_serializing_if = "Vec::is_empty")]
-        trivia: Vec<Rc<Node>>,
-    },
+pub struct RuleNode {
+    pub kind: RuleKind,
+    pub text_len: TextIndex,
```

Comment:
Nit suggestion: What do you think of `length` instead of `text_len`? The type already starts with "Text". Or the full `text_length` if not.

```suggestion
    pub length: TextIndex,
```

Reply:
I think `text_len` is the best choice because:
1. `len` is the rust standard for length methods
2. Just `len` or `length` would conceivably be the number of childred nodes
3. Which is probably why rowan does it.